Added feature to change color while animating

msztech opened the pull request:
Issue #9 Now users can supply startColor and endColor when calling function increment to change the color for the number increment animation.

msztech committed:
Added feature to change color while animating

Now users can supply startColor and endColor when calling function increment to change the color for the number increment animation.

tbaranes commented on Source/IncrementableLabel.swift:
```diff
@@ -150,6 +183,10 @@ extension IncrementableLabel {
             }
         }
+        //Update the color
+        if startColor != nil && endColor != nil{
```
could you unwrap these optionals in the if instead of force unwrap it after?

```swift
if let startColor = startColor, let endColor = endColor {
            textColor = blend(from: startColor, to: endColor, percent: progress / duration)
}
```

tbaranes commented on Source/IncrementableLabel.swift:
```diff
@@ -119,6 +129,29 @@ extension IncrementableLabel {
     }
+    
+    //From https://stackoverflow.com/a/39779603
+    private func blend(from: UIColor, to: UIColor, percent: Double) -> UIColor {
```
could you move this method in an `UIColor` extension ? (potentially in another swift file)

tbaranes commented on Source/IncrementableLabel.swift:
```diff
@@ -85,6 +88,13 @@ extension IncrementableLabel {
         startIncrementation(fromValue: fromValue, toValue: toValue, duration: duration)
     }
+    
+    public func increment(fromValue: Double, toValue: Double, startColor: UIColor, endColor: UIColor, duration: Double = 0.3, completion: Completion? = nil) {
```
A few things here, I think we need to have that new option for all the `increment`
 methods: `increment `, `incrementFromCurrentValue`, `incrementFromZero`

At the same time, I would like to avoid to increase the number of methods to add this option, so what about adding thesee new variables to the existing one as optional, for example:

```swift
    public func increment(fromValue: Double, toValue: Double, startColor: UIColor? = nil, endColor: UIColor? = nil, duration: Double = 0.3, completion: Completion? = nil) {
```

tbaranes reviewed (changes requested):
Thanks you for opening this PR, this is definitely a feature that I'd like to see here! 
I let you a few comments, could you have a look at them?

You will also need to update the CHANGELOG and README following the API updates.

msztech commented:
@tbaranes 👍🏼 Thanks for the suggestions! Modifying the codes now.

msztech committed:
Color Animation Adjustments

- Moved the blend function as an UIColor extension, created a new file.
- Added the variable fromColor and toColor as optionals in three functions.
- Updated the CHANGELOG and the README
- Changed startColor to fromColor, endColor to toColor for consistancy

msztech commented:
Changes committed 😃

Seems that the build failed because it didn't compile the color extension file UIColor+Blend.swift. Could you help me with that? This is my first time to contribute at Github and I really want to help.

tbaranes commented:
Thanks for the changes, it's perfect! 

The reason why travis is failing is because `UIColor+Blend.swift` has been committed but hasn't be added to the project. Can you confirm that the file is listed in the xcodeproj? If yes, you can check if you don't have any local modification left to commit. 
Also, I don't see any Xcodeproj modifications in the files changed, the issue is definitely there 🙃

msztech committed:
Update project.pbxproj

- Added the UIColor+Blend extension file to the Xcode project.

msztech commented:
Great! I have added the file extension to the project and committed a new version.

tbaranes commented:
All good @msztech, thanks again for the PR :)
